app/controllers/application_controller.rb

```diff
@@ -2,9 +2,27 @@ class ApplicationController < ActionController::API
   after_action :set_content_type, except: [:destroy]
   before_action :verify_accept_header
   before_action :verify_content_type
+  before_action :ensure_payload_conforms_to_jsonapi_format, only: [:create, :update]
 
   private
 
+  def prepare_error(error, status = :bad_request)
```

Comment:
Ladies and gentlemen, I don't know whether this is a better implementation over overriding `ApplicationRecord#errors`.
```ruby
def errors
  errors = super
  ...
end
```